src/charm.py

```diff
@@ -157,6 +158,21 @@ def _on_get_issued_certificates(self, event: ActionEvent) -> None:
         results = {"certificates": [certificate.to_json() for certificate in certificates]}
         event.set_results(results)
 
+    def _on_rotate_private_key(self, event: ActionEvent):
+        """Handle the rotate-private-key action.
+
+        Args:
+            event (ActionEvent): Juju event
+        """
+        if not self.unit.is_leader():
+            event.fail("This action can only be run on the leader unit.")
+            return
+        self.tls_certificates.revoke_all_certificates()
+        self._clean_up_juju_secret(EXPIRING_CA_CERTIFICATES_SECRET_LABEL)
+        logger.info("Revoked all previously issued certificates.")
```

Comment:
I'd place this log right after the `revoke_all_certificates` call, now it would show up after any log part of the `_clean up_juju_secret` func.

Reply:
Done